### DISCUSSION TITLE
Docker support

### DISCUSSION THREAD

**Discussion by bst1n** [webstudio-is/webstudio-community#68] [Wishlist]
This would make Webstudio easier to adopt for all those Docker and Portainer users who find it more cumbersome to turn to Vercel or Netlify.

**Comment by rogerznts** [comment]
I managed to install the developer version on my private server, using Docker, but it was quite a journey. When I tried to run the release version, it gave a lot of errors. Only the dev version worked.

**Reply by kof** [reply]
sorry for the experience, we didn't have time to make an easy to use docker yet, self-hosting outside of vercel/netflify requires a lot of technical insight atm.

**Reply by rogerznts** [reply]
I have a docker folder with a DockerFile, which at the root of the project, creates an image with the pnpm installation and enables the entire environment ready for development. In addition to docker-compose.yml with the local volume. I can make a merge request, does it make sense to collaborate?

**Reply by rogerznts** [reply]
Works with docker swarm mode too

**Reply by kof** [reply]
I would prefer if oyu created a repository and we link it in the documentation, so you can keep updating it without us being involved

**Reply by bst1n** [reply]
@kof @rogerznts https://easypanel.io/dockerizer/

**Reply by HiroDane** [reply]
@rogerznts Can you share the Dockerfile. I have lost hours trying to get this anywhere close to working. The "Contributing as a developer" guide seems to be out of date or at least very incomplete.

**Reply by kof** [reply]
Also please send us PRs when you find things in contribution guide that are missing or outdated

**Reply by sycured** [reply]
@rogerznts can you share your setup? setting webstudio up and running is painful and in addition, my last attempt resulted in missing marketplace (empty, no section/page)

**Reply by kof** [reply]
We need to make docker setup easier, so feel free to contribute with whatever can improve docker experience.

As for marketplace - I don't think  it can be part of a self-hostable webstudio. Quite literally marketplace is by definition not your's, items there belong to other people.

**Reply by sycured** [reply]
@kof I hoped to see some "basic/essentials" shipped in the marketplace... but ok, it's strange to have 0 templates/examples available

About the setup I did in a VM:
1. install:
    - PostgreSQL and create a user/password for webstudio
    - nodejs 20
    - pnpm (latest/current)
2. first step
```bash
pnpm config set store-dir $HOME/.pnpm-store
git clone https://github.com/webstudio-is/webstudio.git
cd webstudio
corepak enable
corepak install
sed -i '54d' package.json
pnpm install
```
3. second step
    - edit apps/builder/.env with the correct PostgreSQL username/password + AUTH_TOKEN
    - copy .env to .env.production
    - copy .env to .env.development
5. final step
```bash
pnpm run build
pnpm run dts
cd apps/builder
pnpm migrations migrate --dev
cd ../..
pnpm run dev
```

**Reply by rogerznts** [reply]
@HiroDane @sycured @kof 

https://github.com/rogerznts/webstudio/tree/feature/docker-compose/docker

cd docker
docker compose --build
docker compose up

In docker compose, you can use the command that will run the migrations first, and then just run dev. And don't necessarily use wait-for-it.sh... I left comments in the docker compose file, the possibilities

command: ["sh", "-c", "pnpm migrations migrate --dev"]
command: ["sh", "-c", "pnpm run dev"]
command: ["/wait-for-it.sh", "db", "5432", "pnpm migrations migrate --dev && pnpm run dev"]
command: sleep infinity #to run commands in bash

Added a volume to upload persistent data and image from docker hub

Sorry for not responding sooner, I've been away from development

**Reply by nikola66** [reply]
It seem the repo has been deleted? 404 missing, not working @rogerznts 
https://github.com/rogerznts/webstudio/tree/feature/docker-compose/docker

**Reply by rogerznts** [reply]
yes, it was deleted because it no longer worked with the updates. anyway, I ended up stopping using the tool and moving on to another one, frappe builder, it worked better for me

**Comment by bst1n** [comment]
this is solved right ?

**Reply by kof** [reply]
Yes, for published site. This issue based on comments seems to be a mixed bag between builder and sites. These are basically separate programs.
